pkg/apis/serving/k8s_validation.go

```diff
@@ -341,7 +341,7 @@ func ValidatePodSpec(ctx context.Context, ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
-func validateInitContainers(ctx context.Context, containers []corev1.Container, volumes map[string]corev1.Volume) (errs *apis.FieldError) {
+func validateInitContainers(ctx context.Context, containers, otherContainers []corev1.Container, volumes map[string]corev1.Volume) (errs *apis.FieldError) {
```

Comment:
Would it make sense to have a "top-level" (i.e. directly called in `ValidatePodSpec` before even calling the other validation functions) `validateContainerNames` that takes them all and does the checking? Seems like we'd safe some duplication and maybe would add some additional clarity that way.

Reply:
@markusthoemmes thanks for your reply. I considered doing the check in `validateContainerNames ` or something similar but I finished it [like kubernetes dose](https://github.com/kubernetes/kubernetes/blob/7fbb384e15e2fb8370513eab9443c5107df1b257/pkg/apis/core/validation/validation.go#L2933). which better ?

Reply:
Fine to keep as-is!